routers/web/repo/issue.go

```diff
@@ -1799,6 +1799,27 @@ func GetIssueInfo(ctx *context.Context) {
 		}
 		return
 	}
+
+	if issue.IsPull {
+		// Need to check if Pulls are enabled and we can read Pulls
+		if !ctx.Repo.Repository.CanEnablePulls() || !ctx.Repo.CanRead(unit.TypePullRequests) {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
+	} else {
+		// Need to check if Issues are enabled and we can read Issues
+		if !ctx.Repo.CanRead(unit.TypeIssues) &&
+			!ctx.Repo.CanRead(unit.TypeExternalTracker) {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
+		_, err := ctx.Repo.Repository.GetUnit(unit.TypeExternalTracker)
+		if err == nil {
+			ctx.Error(http.StatusNotFound)
+			return
+		}
```

Comment:
Actually we can be even simpler here.

```suggestion
		if !ctx.Repo.CanRead(unit.TypeIssues)  {
			ctx.Error(http.StatusNotFound)
			return
		}
```